pkg/service/tap_service.go

```diff
@@ -28,9 +30,27 @@ func (tapService *TAPService) DisconnectFromTurbo() {
 	glog.V(4).Infof("[DisconnectFromTurbo] End *********")
 }
 
-// get the jwtToken from the auth component using client id and client secret fed from k8s secret
-// client id and secret are not provided, empty jwtToken will be sent to the channel for websocket connection
-// empty jwtToken will be ignored in websocket
+// Get the jwtToken from the auth component using client id and client secret fed from k8s secret.
+// If client id and secret are not provided, empty jwtToken will be sent to the channel for websocket connection.
+// ConnectToTurbo() will attempt to establish a non-secure websocket connection.
+// Empty jwtToken will be ignored in websocket
+//
+// In order to establish a secure websocket connection with the server,
+// 1. First, a token is obtained by sending a http request to the Hydra service (OAuth Client) using the client id
+// and secret configured for the probe.
+// 2. Then, the hydra token is exchanged for a JWT token by sending a request to the auth service.
+//
+// If the Hydra service returns errors or 401 (invalid credentials), this request is re-tried
+// until a valid token is returned, websocket connection is not established with the server.
```

Comment:
Secret is provided by the probe to turbo-go-sdk/TapService.
Probe users will have to reconfigure/redeploy the correct/right Kubernetes secret and restart the probe.

Reply:
https://github.com/turbonomic/turbo-api/pull/17 added to log the steps necessary to fix the probe connection.
As discussed, a separate effort will be carried out the propagate this error to the probes, so the corrected client id/secret from the Kubernetes secret will be re-fetched by the probe before retrying, to avoid probe restarts.